src/pages/index.js

```diff
@@ -81,35 +143,174 @@ profileEditButton.addEventListener("click", () => {
 const imagePopup = new PopupWithImage("#preview-modal");
 imagePopup.setEventListeners();
 
-// Enable form validation
+// add event listener to avatar
+document
+  .querySelector(".profile__edit-avatar-button")
+  .addEventListener("click", () => {
+    editAvatarModal.open();
+  });
+// when it fires, open avatar image modal
+
+/* ------------------------ Enable form validation ------------------------ */
+
+const editFormValidator = new FormValidator(
+  validationSettings,
+  profileEditForm
+);
+const addFormValidator = new FormValidator(
+  validationSettings,
+  addCardFormElement
+);
+
+const editAvatarValidator = new FormValidator(
+  validationSettings,
+  document.querySelector("#edit-avatar-modal")
+);
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
-
+editAvatarValidator.enableValidation();
 /* -------------------------------------------------------------------------- */
 /*                                 Functions;                                 */
 /* -------------------------------------------------------------------------- */
-function handleImageClick(data) {
-  imagePopup.open(data);
+function handleCardPreview(cardData) {
+  previewImagePopup.open(cardData);
 }
+
+const avatarSubmitButton = document.querySelector("#avatar-submit-button");
+
+// function handleAvatarEditSubmit(input) {
+//   const link = input.avatar;
+//   api.updateAvatar(link);
+//   if (link) {
+//     avatarSubmitButton.textContent = "Saving...";
+//     api
+
+//       .editProfileImage(link)
+//       .then((userData) => {
+//         userInfo.setUserInfo(userData);
+//         editAvatarModal.close();
+//       })
+//       .catch((err) => console.error("Error updating avatar:", err))
+//       .finally(() => {
+//         editAvatarModal.setLoadingState(false);
+//       });
+//   } else {
+//     console.error("Avatar Link is not defined");
+//   }
+// }
+
+function handleAvatarEditSubmit(input) {
+  const link = input.avatar;
+
+  if (link) {
+    // set loading state to true
+    editAvatarModal.setLoadingState(true);
+    // the right api call is updateAvatar
+    api
+      .updateAvatar(link)
+      .then((userData) => {
+        // set avatar info
+        userInfo.setAvatarInfo(userData.avatar);
+
+        editAvatarModal.close();
+      })
+      .catch((err) => console.error("Error updating avatar:", err))
+      .finally(() => {
+        // set loading state to false
+        editAvatarModal.setLoadingState(false);
+      });
+  } else {
+    console.error("Avatar Link is not defined");
+  }
+}
+
 function handleProfileEditSubmit(formValues) {
   userInfo.setUserInfo({
```

Comment:
Every server request in `index.js` should look something like this:

```js
api.someMethod()
.then(res => {
// here is the response that should be used to update the DOM.
// also in `then` you should close the popup
})
.catch(err => // here you catch possible errors)
.finally(() => // here you return the default button text back )
``` 

Please, correct it in the whole code